src/win/pty.cc

```diff
@@ -196,10 +205,7 @@ static NAN_METHOD(PtyStartProcess) {
   winpty_error_ptr_t error_ptr = nullptr;
   winpty_config_t* winpty_config = winpty_config_new(0, &error_ptr);
   if (winpty_config == nullptr) {
-    std::wstring msg(winpty_error_msg(error_ptr));
-    std::string msg_(msg.begin(), msg.end());
-    why << "Error creating WinPTY config: " << msg_;
-    Nan::ThrowError(why.str().c_str());
+    throw_winpty_error("Error creating WinPTY config", error_ptr);
     goto cleanup;
   }
   winpty_error_free(error_ptr);
```

Comment:
I think the code leaks error objects when an error occurs.  Should there should be another `winpty_error_free` call for the `cleanup` label?  (If you add one, make sure you don't call `winpty_error_free` twice on the same non-nullptr value -- that would cause a double-free crash.  Maybe set `error_ptr` to `nullptr` after each free call.)

The other calls could be removed, in practice, because each winpty API either:

 * succeeds, and sets *error_ptr to nullptr, OR
 * fails, and sets *error_ptr to an error object.

Calling `winpty_error_free` on `nullptr` has no effect.  I'm not sure whether the best convention is to always call `winpty_error_free` after each API call, or whether to only call it after detecting an error.